src/components/BudgetController.js

```diff
@@ -20,17 +20,15 @@ function Expense(id, description, value) {
   this.percentage = -1;
 }
 
-Expense.prototype.calcPercentage = function(totalIncome) {
+Expense.prototype.calcPercentage = totalIncome => {
```

Comment:
The application is failing due to the arrow function here. There is a _this_ inside the arrow function and the scope is pointing to the window object instead of the object.